qiskit_optimization/algorithms/optimization_algorithm.py

```diff
@@ -196,6 +196,26 @@ def get_correlations(self) -> np.ndarray:
                         correlations[i, j] -= prob
         return correlations
 
+    def get_probabilities(self, status=None) -> Dict[str, float]:
+        """
+        Returns samples, and function values that are successful.
+
+        Returns:
+            A dictionary with sample names, and function values.
+        """
+        sample_results = {}
+        if status is None:
+            for sample in self.samples:
+                sample_values = f"{''.join(map(str, map(int, sample.x)))}"
+                sample_results.update({sample_values: sample.probability})
+        else:
+            for sample in self.samples:
+                sample_values = f"{''.join(map(str, map(int, sample.x)))}"
+                if sample.status.name in status:
```

Comment:
You can simplify the code by checking the condition as follows.
```
if status is None or sample.status == status:
```
As for `status` arg, it should be `Optional[OptimizationResultStatus]`. We need to take advantage of enum.

Reply:
Thanks for the tip about condition.

How can we use `Optional[OptimizationResultStatus]` to choose the `status` that we're interested in, like in example below with current behavior.

![image](https://user-images.githubusercontent.com/42217706/120189921-38083e00-c218-11eb-802d-a6720afe2f0a.png)